### PR TITLE
 Split operation response types that we had over-generalized

### PR DESCRIPTION
**Pull request by tg-msft**
Split a few swagger types into unique responses.

The only changes for this PR are in the last commit.

### CODE REVIEW

**Review by kfarmer-msft — APPROVED** [review]
Thought: Maybe a matrix showing response type vs API?  I wonder if that would help with aligning mental models about how things work.

File: sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs [hunk 1]
```diff
@@ -548,14 +548,14 @@ protected sealed override BlobBaseClient WithSnapshotImpl(string snapshot)
         /// notifications that the operation should be cancelled.
         /// </param>
         /// <returns>
-        /// A <see cref="Response{BlobContentInfo}"/> describing the
-        /// state of the updated block blob.
+        /// A <see cref="Response{BlockInfo}"/> describing the
```
Comment (maririos) [inline sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs]: I imagine there was a conversation about this type name `BlockInfo`? usually type names are complete and not abbreviations. Also, `Info` doesn't help telling the user what the type is or what type of content it has...

**Review by maririos — APPROVED** [review]
Besides the comment on the type names, looks good

File: sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs (hunk 1 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs]: We've been using `Info` all over the place for our Response types.  This is one of those "okay for Preview" gray areas the Arch Board wants to lock down before the next release.

**Pull request merged**